.github/workflows/cd.yml

```
@@ -0,0 +1,38 @@
+name: cd
+on:
+  workflow_run:
+    workflows: ["ci"]
+    types:
+    - completed
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: cd/checkout
+      uses: actions/checkout@dc323e67f16fb5f7663d20ff7941f27f5809e9b6 # v3.2.0
+
+    - name: cd/set-tag-pr
+      if: github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.conclusion == 'success'
+      run: echo "IMAGE_TAG=${GITHUB_SHA:0:7}" >> $GITHUB_ENV
+
+    - name: cd/set-tag-push
+      if: github.event.workflow_run.event == 'push' && github.event.workflow_run.conclusion == 'success' && startsWith(github.event.workflow_run.head_ref, 'refs/tags/')
```

This payload does not have a `head_ref` directive. Check [here](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_run)

Check github context https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

```suggestion
      if: github.ref_type == 'tag' 
```